### DISCUSSION TITLE
Using @apply in JS ( detecting directives via JIT/watch )

### DISCUSSION THREAD

**Discussion by drschwabe** [tailwindlabs/tailwindcss#5131] [Help]
I have basic tailwind.config.js: 

```js
module.exports = {
  mode: 'jit',
  purge: ['serve.js'],
//etc... 
```

Where my 'serve.js' file contains a variable with a template literal string that has all my HTML and tailwind classes in it. 

The JIT/tailwind watch CLI tool is working good to scan my serve.js file and created a purge'ed output CSS file.  

However, I want to use the @apply directive

but I don't think that is necessarily supported or documented in this context.   It doesn't seem like the JIT/watch CLI command is detecting if I put @apply directives in that file (I tried putting one in a <style> tag within my main HTML template string for example).    

Is there a way to 'trigger' those to work as intended ?

**Comment by simonswiss** [comment]
Hey!

Do you have a code example to share? If your `@apply` declarations are the CSS file that gets processed through PostCSS, they should be "acknowledged" and not purged.

You might need to wrap these in an `@layer` block though:

<img src="https://user-images.githubusercontent.com/485747/127794133-c1e269fa-d00e-48d5-b176-0b25c8b76a99.png" alt="" 
 width="680" />

https://tailwindcss.com/docs/just-in-time-mode#limitations
